src/main/kotlin/me/partlysanestudios/partlysaneskies/events/EventManager.kt

```diff
@@ -41,19 +45,19 @@ object EventManager {
     fun onChatReceivedEvent(event: ClientChatReceivedEvent) {
         if (event.type.toInt() != 0) return
 
-        if (PSSChatEvent(event.message.formattedText, event.message).post()) {
+        if (PSSChatEvent(event.message.formattedText, event.message).apply { if (event.isCanceled) cancel() }.post()) {
```

Comment:
```suggestion
        if (PSSChatEvent(event.message.formattedText, event.message).post()) {
```

thats not how it works, cancel() alr sets isCanceled to true
and even if, isCanceled cant be true before we even post the event